### PR TITLE
RELATED: ONE-4621 - Enhance backend-spi to include 'ref' props in dimension descriptors

### PR DESCRIPTION
**Pull request by lupko**
This is required to enable backend-agnostic processing of dimension descriptors.

The recording infrastructure is augmented so that it does not store the 'ref' and instead dynamically injects the 'ref' properties in the recordedBackend implementation - according to the backend preference.

---

Supported PR commands:

| Command         | Description            |
| --------------- | ---------------------- |
| `ok to test`    | Re-run standard checks |
| `extended test` | BackstopJS tests       |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

File: libs/sdk-backend-tiger/src/convertors/fromBackend/afm/dimensions.ts [hunk 1]
```diff
@@ -54,6 +57,10 @@ function transformDimension(dim: Execution.IResultDimension): IDimensionDescript
                                     localIdentifier: m.measureHeaderItem.localIdentifier,
                                     name: m.measureHeaderItem.name ?? m.measureHeaderItem.localIdentifier,
                                     format: m.measureHeaderItem.format ?? DEFAULT_FORMAT,
+                                    identifier: m.measureHeaderItem.identifier,
+                                    ref: m.measureHeaderItem.identifier
+                                        ? idRef(m.measureHeaderItem.identifier, "measure")
```
Comment (no23reason) [inline libs/sdk-backend-tiger/src/convertors/fromBackend/afm/dimensions.ts:62]: Can this ever be a `"fact"`?

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: libs/sdk-backend-tiger/src/convertors/fromBackend/afm/dimensions.ts (hunk 1 above)
Comment (lupko) [reply libs/sdk-backend-tiger/src/convertors/fromBackend/afm/dimensions.ts:62]: good catch. looking into tiger backend code, i see it does not even return identifier. to match the SPI contract, i'm going to augment the descriptor with identifier & ref used on the execution definition.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/659/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2246/ : SUCCESS in 11m 46s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/305/ : SUCCESS in 31s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/561/ : SUCCESS in 16s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2250/ : SUCCESS in 16m 29s

**Pull request merged**